proof/crefine/AARCH64/VSpace_C.thy

```diff
@@ -3047,6 +3047,164 @@ lemma vcpu_switch_ccorres:
         (vcpuSwitch v) (Call vcpu_switch_'proc)"
   by (cases v; clarsimp simp: vcpu_switch_ccorres_None[simplified] vcpu_switch_ccorres_Some[simplified])
 
+lemma modify_armHSCurVCPU_split:
+  "modifyArchState (armHSCurVCPU_update (\<lambda>_. p)) = do modifyArchState (armHSCurVCPU_update f);
+                                                      modifyArchState (armHSCurVCPU_update (\<lambda>_. p))
+                                                   od"
+  apply (clarsimp simp: modifyArchState_def modify_modify)
+  apply (rule ext)
```

Comment:
Have you tried the `monad_eq` tactic for these? Might not be strong enough, but sometimes it is surprisingly good.

Reply:
This and the following lemmas were actually just moved from a later file. I'll do the easier cleanup from your suggestions but might leave the more involved changes for another time.

Reply:
No worries